Xamarin.Forms.Core/Popup/BasePopup.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+
+namespace Xamarin.Forms
+{
+	public abstract class BasePopup : View
```

Comment:
Do you have any reason inherit from `View`? I think it should be inherit `Element` instead `View` because it is not `View`, it can't layouting by `Layout` class

Reply:
The `BasePopup` needs to inherit from `View` because it has layout context which allows you to place the `Popup` in 9 positions on the screen
* top left
* top center
* top right
* right middle
* bottom right
* bottom center
* bottom left
* left middle
* center

The `View` object that is encapsulated in the `BasePopup` is used for rendering the content that is rendered inside of the native popup control. This content also has layout options within the context of the native popup which is why we require the Layout Options for both the native popup and the content that is rendered on the inside of the popup.

**When I work on my next push for this PR I am going to double check all of this to make sure this is how it is working.** If I can update this to `Element` I'll make the change. Thanks for the feedback 👍

Reply:
hm, if BasePopup was inherit from View, It can be contained by Layout class, but it seems not allow it, 
 for eaxmple, it is progrmatically allowed code, but actually, it is not working
``` c#
  new StackLayout {
       Children = {
              new Popup {
                   View = new Label { ... }
              }
  }
```

Reply:
I see your point now, thanks for clarifying. The Popup should not be used like that and if it is inherited from `View`, it will compile to be used like that. I agree with you 100% now and it doesn't make sense for `BasePopup` to inherit from `View`. I will take another look at this and use `Element` or similar parent class on my next set of work for this PR.

Reply:
I have updated the `BasePopup` to inherit from `VisualElement`